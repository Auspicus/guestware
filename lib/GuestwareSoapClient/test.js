var GuestwareSoapClient = require('./index.js');
var requestReadData = require('./read.json');
var requestUpdateData = require('./update.json');

GuestwareSoapClient.prototype.read = function (method, args) {
  return new Promise(function (resolve, reject) {
    resolve(this.mockResponse(method, requestReadData[method]));
  }.bind(this));
}

GuestwareSoapClient.prototype.update = function (method, objectName, changeList) {
  return new Promise(function (resolve, reject) {
    resolve(this.mockResponse(method, requestUpdateData[method]));
  }.bind(this));
}

describe('# GuestwareSoapClient', function () {
  var instance = new GuestwareSoapClient(
    'http://example.com/wsdl',
    'TestAppName',
    'TestVersionNumber',
    'TestAppId',
    'TestUsername',
    'TestPassword'
  );

  describe('Request Generation', function () {
    it('should generate a valid read request header', function () {
      expect(instance.requestReadHeader).toMatchSnapshot();
    });

    it('should generate a valid read request footer', function () {
      expect(instance.requestReadFooter).toMatchSnapshot();
    });

    it('should generate a valid update request header', function () {
      expect(instance.requestUpdateHeader).toMatchSnapshot();
    });

    it('should generate a valid update request footer', function () {
      expect(instance.requestUpdateFooter).toMatchSnapshot();
    });

    it('should generate a valid read request', function () {
      expect(instance.generateReadRequestBody('ReadTestRequestMethod', {
        parintExampleArgument: 1
      })).toMatchSnapshot();
    });

    it('should generate a valid update request', function () {
      expect(instance.generateUpdateRequestBody(
        'UpdateTestRequestMethod',
        'pardstTestComplexType',
        [{
          $$elementType: 'TEST_ELEMENT',
          $$elementUpdateType: 'modified',
          ExampleAttribute: 'ExampleValue',
          TestAttribute: 1
        }]
      )).toMatchSnapshot();
    })
  });

  describe('Response', function () {
    it('should return a response from a valid read request', function () {
      instance
      .read('ReadGuestLogin', { parintGuestID: 0 })
      .then(function (response) {
        expect(response.raw).toMatchSnapshot();
      })
      .catch(function (err) {
        throw err;
      })
    });

    it('should return an error from an invalid read request', function () {

    });

    it('should return a response from a valid update request', function () {

    });

    it('should return an error from an invalid update request', function () {

    });
  });
});
