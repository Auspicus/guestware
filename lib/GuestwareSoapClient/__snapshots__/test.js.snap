// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# GuestwareSoapClient Request Generation should generate a valid read request 1`] = `
Object {
  "raw": Array [
    "<soapenv:Envelope xmlns:soapenv=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:web=\\"http://webservices.guestware.com/\\">",
    "<soapenv:Header>",
    "<web:TestAppId>",
    "<web:UserName>TestUsername</web:UserName>",
    "<web:PassWord>TestPassword</web:PassWord>",
    "<web:ApplicationName>TestAppName</web:ApplicationName>",
    "<web:VersionNumber>TestVersionNumber</web:VersionNumber>",
    "</web:TestAppId>",
    "</soapenv:Header>",
    "<soapenv:Body>",
    "<web:ReadTestRequestMethod>",
    "<web:parintExampleArgument>",
    1,
    "</web:parintExampleArgument>",
    "</web:ReadTestRequestMethod>",
    "</soapenv:Body>",
    "</soapenv:Envelope>",
  ],
  "string": "<soapenv:Envelope xmlns:soapenv=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:web=\\"http://webservices.guestware.com/\\"><soapenv:Header><web:TestAppId><web:UserName>TestUsername</web:UserName><web:PassWord>TestPassword</web:PassWord><web:ApplicationName>TestAppName</web:ApplicationName><web:VersionNumber>TestVersionNumber</web:VersionNumber></web:TestAppId></soapenv:Header><soapenv:Body><web:ReadTestRequestMethod><web:parintExampleArgument>1</web:parintExampleArgument></web:ReadTestRequestMethod></soapenv:Body></soapenv:Envelope>",
}
`;

exports[`# GuestwareSoapClient Request Generation should generate a valid read request footer 1`] = `
Array [
  "</soapenv:Body>",
  "</soapenv:Envelope>",
]
`;

exports[`# GuestwareSoapClient Request Generation should generate a valid read request header 1`] = `
Array [
  "<soapenv:Envelope xmlns:soapenv=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:web=\\"http://webservices.guestware.com/\\">",
  "<soapenv:Header>",
  "<web:TestAppId>",
  "<web:UserName>TestUsername</web:UserName>",
  "<web:PassWord>TestPassword</web:PassWord>",
  "<web:ApplicationName>TestAppName</web:ApplicationName>",
  "<web:VersionNumber>TestVersionNumber</web:VersionNumber>",
  "</web:TestAppId>",
  "</soapenv:Header>",
  "<soapenv:Body>",
]
`;

exports[`# GuestwareSoapClient Request Generation should generate a valid update request 1`] = `
Object {
  "raw": Array [
    "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>",
    "<soap:Envelope xmlns:soap=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:xsd=\\"http://www.w3.org/2001/XMLSchema\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\">",
    "<soap:Header>",
    "<TestAppId xmlns=\\"http://webservices.guestware.com/\\">",
    "<UserName>TestUsername</UserName>",
    "<PassWord>TestPassword</PassWord>",
    "</TestAppId>",
    "</soap:Header>",
    "<soap:Body>",
    "<UpdateTestRequestMethod xmlns=\\"http://webservices.guestware.com/\\">",
    "<pardstTestComplexType xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\" msdata:SchemaSerializationMode=\\"ExcludeSchema\\">",
    "<xs:schema xmlns:xs=\\"http://www.w3.org/2001/XMLSchema\\" xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns:msprop=\\"urn:schemas-microsoft-com:xml-msprop\\" xmlns:mstns=\\"http://webservices.guestware.com/dstGST.xsd\\" id=\\"dstGST\\" targetNamespace=\\"http://webservices.guestware.com/dstGST.xsd\\" attributeFormDefault=\\"qualified\\" elementFormDefault=\\"qualified\\">",
    "<xs:element name=\\"dstGST\\" msdata:IsDataSet=\\"true\\" msdata:UseCurrentLocale=\\"true\\">",
    "<xs:complexType>",
    "<xs:choice minOccurs=\\"0\\" maxOccurs=\\"unbounded\\"/>",
    "</xs:complexType>",
    "</xs:element>",
    "</xs:schema>",
    "<diffgr:diffgram xmlns:diffgr=\\"urn:schemas-microsoft-com:xml-diffgram-v1\\">",
    "<dstGST xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\">",
    "<TEST_ELEMENT diffgr:id=\\"TEST_ELEMENT1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"modified\\">",
    "<ExampleAttribute>ExampleValue</ExampleAttribute>",
    "<TestAttribute>1</TestAttribute>",
    "</TEST_ELEMENT>",
    "</dstGST>",
    "<diffgr:before>",
    "<TEST_ELEMENT diffgr:id=\\"TEST_ELEMENT1\\" msdata:rowOrder=\\"0\\">",
    "<ExampleAttribute>ExampleValue</ExampleAttribute>",
    "<TestAttribute>1</TestAttribute>",
    "</TEST_ELEMENT>",
    "</diffgr:before>",
    "</diffgr:diffgram>",
    "</pardstTestComplexType>",
    "</UpdateTestRequestMethod>",
    "</soap:Body>",
    "</soap:Envelope>",
  ],
  "string": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><soap:Envelope xmlns:soap=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:xsd=\\"http://www.w3.org/2001/XMLSchema\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\"><soap:Header><TestAppId xmlns=\\"http://webservices.guestware.com/\\"><UserName>TestUsername</UserName><PassWord>TestPassword</PassWord></TestAppId></soap:Header><soap:Body><UpdateTestRequestMethod xmlns=\\"http://webservices.guestware.com/\\"><pardstTestComplexType xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\" msdata:SchemaSerializationMode=\\"ExcludeSchema\\"><xs:schema xmlns:xs=\\"http://www.w3.org/2001/XMLSchema\\" xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns:msprop=\\"urn:schemas-microsoft-com:xml-msprop\\" xmlns:mstns=\\"http://webservices.guestware.com/dstGST.xsd\\" id=\\"dstGST\\" targetNamespace=\\"http://webservices.guestware.com/dstGST.xsd\\" attributeFormDefault=\\"qualified\\" elementFormDefault=\\"qualified\\"><xs:element name=\\"dstGST\\" msdata:IsDataSet=\\"true\\" msdata:UseCurrentLocale=\\"true\\"><xs:complexType><xs:choice minOccurs=\\"0\\" maxOccurs=\\"unbounded\\"/></xs:complexType></xs:element></xs:schema><diffgr:diffgram xmlns:diffgr=\\"urn:schemas-microsoft-com:xml-diffgram-v1\\"><dstGST xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\"><TEST_ELEMENT diffgr:id=\\"TEST_ELEMENT1\\" msdata:rowOrder=\\"0\\" diffgr:hasChanges=\\"modified\\"><ExampleAttribute>ExampleValue</ExampleAttribute><TestAttribute>1</TestAttribute></TEST_ELEMENT></dstGST><diffgr:before><TEST_ELEMENT diffgr:id=\\"TEST_ELEMENT1\\" msdata:rowOrder=\\"0\\"><ExampleAttribute>ExampleValue</ExampleAttribute><TestAttribute>1</TestAttribute></TEST_ELEMENT></diffgr:before></diffgr:diffgram></pardstTestComplexType></UpdateTestRequestMethod></soap:Body></soap:Envelope>",
}
`;

exports[`# GuestwareSoapClient Request Generation should generate a valid update request footer 1`] = `
Array [
  "</soap:Body>",
  "</soap:Envelope>",
]
`;

exports[`# GuestwareSoapClient Request Generation should generate a valid update request header 1`] = `
Array [
  "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>",
  "<soap:Envelope xmlns:soap=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:xsd=\\"http://www.w3.org/2001/XMLSchema\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\">",
  "<soap:Header>",
  "<TestAppId xmlns=\\"http://webservices.guestware.com/\\">",
  "<UserName>TestUsername</UserName>",
  "<PassWord>TestPassword</PassWord>",
  "</TestAppId>",
  "</soap:Header>",
  "<soap:Body>",
]
`;

exports[`# GuestwareSoapClient Response should return a response from a valid read request 1`] = `"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?><soap:Envelope xmlns:soap=\\"http://schemas.xmlsoap.org/soap/envelope/\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\" xmlns:xsd=\\"http://www.w3.org/2001/XMLSchema\\"><soap:Body><ReadGuestLoginResponse xmlns=\\"http://webservices.guestware.com/\\"><ReadGuestLoginResult msdata:SchemaSerializationMode=\\"ExcludeSchema\\" xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\"><xs:schema id=\\"dstGST\\" targetNamespace=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns:mstns=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\" xmlns:xs=\\"http://www.w3.org/2001/XMLSchema\\" xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\" xmlns:msprop=\\"urn:schemas-microsoft-com:xml-msprop\\" attributeFormDefault=\\"qualified\\" elementFormDefault=\\"qualified\\"><xs:element name=\\"dstGST\\" msdata:IsDataSet=\\"true\\" msdata:UseCurrentLocale=\\"true\\"><xs:complexType><xs:choice minOccurs=\\"0\\" maxOccurs=\\"unbounded\\" /></xs:complexType></xs:element></xs:schema><diffgr:diffgram xmlns:msdata=\\"urn:schemas-microsoft-com:xml-msdata\\" xmlns:diffgr=\\"urn:schemas-microsoft-com:xml-diffgram-v1\\"><dstGST xmlns=\\"http://webservices.guestware.com/dstGST.xsd\\"><GUEST_LOGIN diffgr:id=\\"GUEST_LOGIN1\\" msdata:rowOrder=\\"0\\"><GuestID>0</GuestID><GuestLoginID>email@example.com</GuestLoginID><GuestLoginPassword>supersecurepassword</GuestLoginPassword><CultureID>en</CultureID><LogActivity>false</LogActivity><DisableLogin>false</DisableLogin><EntryDate>2017-07-23T20:59:41.481Z</EntryDate><EntryBy>GWAdminAccountName</EntryBy><LastEditDate>2017-07-23T20:59:41.481Z</LastEditDate><LastEditBy>GWAdminAccountName</LastEditBy></GUEST_LOGIN></dstGST></diffgr:diffgram></ReadGuestLoginResult></ReadGuestLoginResponse></soap:Body></soap:Envelope>"`;
